text/profiles/0237-profiles-data-model.md

```diff
@@ -0,0 +1,1015 @@
+  uint32 unit_index = 4;
+
+  // List of indices referring to Stacktraces in the Profile's stacktrace table.
+  repeated uint32 stacktrace_indices = 10;
+
+  // List of indices referring to Links in the Profile's link table. Each link corresponds to a Stacktrace in stacktrace_indices list. Length must match stacktrace_indices length. [Optional]
+  repeated uint32 link_indices = 11;
+
+  // List of indices referring to AttributeSets in the Profile's attribute set table. Each attribute set corresponds to a Stacktrace in stacktrace_indices list. Length must match stacktrace_indices length. [Optional]
+  repeated uint32 attribute_set_indices = 12;
+
+  // List of values. Each value corresponds to a Stacktrace in stacktrace_indices list. Length must match stacktrace_indices length.
+  repeated int64 values = 13;
+
+  // List of timestamps. Each timestamp corresponds to a Stacktrace in stacktrace_indices list. Length must match stacktrace_indices length.
+  repeated uint64 timestamps = 14;
+}
+
+// SymbolFidelity enumerates the level of fidelity for symbol information in a profile.
+enum SymbolFidelity {
+  SYMBOL_FIDELITY_UNSPECIFIED = 0; // The symbol fidelity level is unspecified.
+  SYMBOL_FIDELITY_FULL = 1;        // The symbol fidelity is at its fullest level of detail.
+}
+
+// Describes the mapping from a binary to its original source code. These are stored in a lookup table in a Profile. These are referenced by index from other messages.
+message Mapping {
+  // Address at which the binary (or DLL) is loaded into memory.
+  uint64 memory_start = 1;
+  // The limit of the address range occupied by this mapping.
+  uint64 memory_limit = 2;
+  // Offset in the binary that corresponds to the first mapped address.
+  uint64 file_offset = 3;
+  // The object this entry is loaded from.  This can be a filename on
+  // disk for the main binary and shared libraries, or virtual
+  // abstractions like "[vdso]". Index into string table
+  uint32 filename_index = 4;
+  // A string that uniquely identifies a particular program version
+  // with high probability. E.g., for binaries generated by GNU tools,
+  // it could be the contents of the .note.gnu.build-id field. Index into string table
+  uint32 build_id_index = 5;
+
+  SymbolFidelity symbolic_info = 6;
+
+  // Reference to an attribute set from the Profile's attribute set table.
+  uint32 attribute_set_index = 7;
+}
+
+// Describes function and line table debug information. These are stored in a lookup table in a Profile. These are referenced by index from other messages.
+message Location {
+  // The id of the corresponding profile.Mapping for this location.
+  // It can be unset if the mapping is unknown or not applicable for
+  // this profile type.
+  uint32 mapping_index = 1;
+  // The instruction address for this location, if available.  It
+  // should be within [Mapping.memory_start...Mapping.memory_limit]
+  // for the corresponding mapping. A non-leaf address may be in the
```

Comment:
Do we need to specify that for leaf address this is the address of the beginning of the instruction? (assuming that is the case).